packages/apputils/src/sessioncontext.tsx

```diff
@@ -1428,20 +1443,40 @@ export class SessionContextDialogs implements ISessionContext.IDialogs {
       buttons: [
         Dialog.cancelButton({ ariaLabel: trans.__('Cancel Kernel Restart') }),
         restartBtn
-      ]
+      ],
+      checkbox: {
+        label: trans.__('Do not ask me again.'),
+        caption: trans.__(
+          'If checked, no confirmation to restart the kernel will be asked in the future.'
+        )
+      }
     });
 
     if (kernel.isDisposed) {
       return false;
     }
     if (result.button.accept) {
+      if (result.isChecked === skipKernelRestartDialogSetting) {
```

Comment:
You need to test if `isChecked` is a boolean because, if the user click on cancel or outside the dialog the checkbox value will be undetermined.

```suggestion
      if (typeof result.isChecked === 'boolean' && result.isChecked === skipKernelRestartDialogSetting) {
```